extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -363,7 +363,10 @@ AnnotationsTransformerBuildItem annotationTransformer(BeanArchiveIndexBuildItem
                 stereotypeAnnotations.add(annotationClass.name());
             }
         }
-        List<DotName> healthAnnotations = new ArrayList<>(5);
+        // TODO additional stereotypes? see AutoProducerMethodsProcessor.annotationTransformer
```

Comment:
Could you be more specific here?

Reply:
I (almost accidentally) stumbled upon this code in the SmallRye Health extension which automatically adds `@Singleton` to classes or methods that don't declare a scope but do declare a MicroProfile Health annotation. This was added by certain @mkouba in https://github.com/quarkusio/quarkus/pull/9749 :-)

To figure out if the class or method in question declares a scope or not, it uses `CustomScopeAnnotationsBuildItem.isScopeIn()` with 1. the set of annotations on the class or method, 2. the set of (meta-)annotations on anything annotated `@Stereotype`. This ignores custom stereotypes, which is what the TODO I added here highlights.

Thinking about it again, I have no idea why `SmallRyeHealthProcessor.annotationTransformer` even exists. `SmallRyeHealthProcessor.build` registers MP Health annotations as bean defining annotations -- except it doesn't define a default scope. We could just define a default scope of `@Singleton` for those additional BDAs and call it a day, no?

Reply:
I tried defining the default scope for the additional BDAs and removing the `annotationTransformer` method and all tests in the SmallRye Health extension pass. I claim that's good enough :-)

Reply:
> Thinking about it again, I have no idea why SmallRyeHealthProcessor.annotationTransformer even exists. 

Most probably because the `BeanDefiningAnnotationBuildItem` originally did not have the `defaultScope` and seems to be implemented on top of stereotypes... which is a bit weird but works sooo.

Reply:
Yea I need to check why the SmallRye Health test failed in CI, it all passed locally.